heft-plugins/heft-storybook-plugin/src/StorybookRunner.ts

```diff
@@ -29,6 +30,12 @@ export class StorybookRunner extends SubprocessRunnerBase<IStorybookRunnerConfig
 
     terminal.writeLine('Launching ' + this._configuration.resolvedStartupModulePath);
 
+    const originalArgv: string[] = process.argv;
+    if (this._configuration.staticBuildOutputDir) {
+      process.argv.push(`--output-dir=${this._configuration.staticBuildOutputDir}`);
```

Comment:
It looks like this will get passed regardless of whether or not the static build module was specified, since the options coalesce into the `resolvedStartupModulePath` option. Does this impact the output folder when specifying `startupModulePath` and running in `--serve` mode? If so, the option is a bit oddly named, and should probably be generalized to `outputDir` or `outputPath`.

Reply:
I pulled your branch and tried it out. Using the setting actually breaks the `--serve` mode flow:
```
---- Bundle started ---- 
[heft-storybook] Starting Storybook runner...
[heft-storybook-runnner] Launching D:\Work\rushstack2\common\temp\node_modules\.pnpm\@storybook+react@6.4.22_6pxnn2xtr5yph4wackblrq76km\node_modules\@storybook\react\bin\index.js
error: unknown option '--output-dir=dist-storybook'
 ---- Bundle continuing with errors (1648ms) ----
-------------------- Finished (7.461s) --------------------
Project: heft-storybook-react-tutorial@1.0.0
Heft version: 0.48.9
Node version: v14.19.1
Error: Subprocess exited before sending "exit" message.
```

You will need to only add this option when running the static build flow.

Reply:
updated, thanks for pointing it out.